rubrics/api/src/main/java/org/sakaiproject/rubrics/api/beans/CriterionTransferBean.java

```diff
@@ -42,4 +42,14 @@ public CriterionTransferBean(Criterion criterion) {
         weight = criterion.getWeight();
         ratings = criterion.getRatings().stream().map(RatingTransferBean::new).collect(Collectors.toList());
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || ((CriterionTransferBean) o).getId() == null || id == null) {
```

Comment:
For what I need, if both id's are null none of them are on database so even if the other fields match you can't consider them the same criterion.
Same with the other answer, ratings might be different but if the number of them within an existing criterion hasn't changed then it can be considered the same criterion (other ratings field might be different).

Reply:
Okay :+1: